tests/pipelines/test_execution_utils.py

```diff
@@ -207,7 +207,7 @@ def name(self):
         def environment(self):
```

Comment:
Can we also add test coverage for the new cached step printing functionality?

Reply:
@liangz1 Awesome stuff! Now that the implementation structure looks good, reminder to add tests